swat/cas/results.py

```diff
@@ -447,9 +447,11 @@ def get_group(_self_, *name, **kwargs):
 
         out = CASResults()
 
-        def set_bykey(attrs, bykey=[]):
```

Comment:
This mutable argument was actually done on purpose.  It is used this way so that the list of By keys only gets done once.  Once the list is populated, it gets reused every time after that.  I did accept the pull request, but I changed this code slightly and put the `bykey` variable outside of the function signature.